### ISSUE TITLE
anyone willing to help me out?

### ISSUE THREAD

**Issue by spormeon** [requirejs/requirejs#1796]
Hi,  this isnt a issue as such,  I'm looking for someone to help me out on little project and dont know where else to post this.  I have a little project that i've split bits of code into 30 odd seperate files and i'm trying to get them to all load properly, somethimes it seems to load propely ( about 75% of the time), other times it doesnt and i cant get my head around it, so looking for some help.

this is my main.js file bt its doesnt always load each file/ module in order, which it needs to do, to make the "whole" thing work:

require(["prebid", "gpt", "preconnect", "bidfilter", "timeoutmap", "floorpricemap", "siteconfig", "unrulyconfig", "amazonbidder", "bidcachemap", "assertiveconfig", "nativebidders", "outstreambidders", "bannerbidders", "adunitsconfig", "custombuckets", "prebidconfig", "assertivecustom", "assertive"], function (prebid, gpt, preconnect, bidfilter, timeoutmap, floorpricemap, siteconfig, unrulyconfig, amazonbidder, bidcachemap, assertiveconfig, nativebidders, outstreambidders, bannerbidders, adunitsconfig, custombuckets, prebidconfig, assertivecustom, assertive) {
});

**Comment by ddziaduch** [comment]
Hi @spormeon it is possible but like that:

```js
require(["prebid"], function (prebid) {
require(["gpt"], function (gpt) {
require(["preconnect"], function (preconnect) {
require(["bidfilter"], function (bidfilter) {
require(["timeoutmap"], function (timeoutmap) {
require(["floorpricemap"], function (floorpricemap) {
require(["siteconfig"], function (siteconfig) {
require(["unrulyconfig"], function (unrulyconfig) {
require(["amazonbidder"], function (amazonbidder) {
require(["bidcachemap"], function (bidcachemap) {
require(["assertiveconfig"], function (assertiveconfig) {
require(["nativebidders"], function (nativebidders) {
require(["outstreambidders"], function (outstreambidders) {
require(["bannerbidders"], function (bannerbidders) {
require(["adunitsconfig"], function (adunitsconfig) {
require(["custombuckets"], function (custombuckets) {
require(["prebidconfig"], function (prebidconfig) {
require(["assertivecustom"], function (assertivecustom) {
require(["assertive"], function (assertive) {
    console.log('all deps loaded in order');
});
});
});
});
});
});
});
});
});
});
});
});
});
});
});
});
});
});
});
```
But this is not recommended way. Please use `shim` or just define dependencies in the modules itself

**Comment by spormeon** [comment]
@ddziaduch Thanks alot,  its seems to work but the console doesn't log any info to console.

would you be willing to have a look at what im doing and see if im going in the right direct etc with it?

**Comment by ddziaduch** [comment]
Do you have any errors? Please check network panel in the dev tools to see whether all files load and whether are loaded in the correct order

**Comment by spormeon** [comment]
i get this big chunk of errror:
script.js:1 Uncaught Error: Mismatched anonymous define() module: function(){function h(a){var c=a._util;c.images=[].slice.call(document.querySelectorAll(a.options.selector));c.count=c.images.length;c.destroyed&&(c.destroyed=!1,a.options.container&&k(a.options.container,function(a){l(a,"scroll",c.validateT)}),l(window,"resize",c.saveViewportOffsetT),l(window,"resize",c.validateT),l(window,"scroll",c.validateT));f(a)}function f(a){for(var c=a._util,b=0;b<c.count;b++){var d=c.images[b],g=d.getBoundingClientRect();if(g.right>=e.left&&g.bottom>=e.top&&g.left<=e.right&&g.top<=e.bottom||-1!==(" "+d.className+" ").indexOf(" "+a.options.successClass+" "))a.load(d),c.images.splice(b,1),c.count--,b--}0===c.count&&a.destroy()}function t(a,c,b){if(c||0<a.offsetWidth&&0<a.offsetHeight)if(c=a.getAttribute(n)||a.getAttribute(b.src)){c=c.split(b.separator);var d=c[p&&1<c.length?1:0];c=new Image;k(b.breakpoints,function(b){a.removeAttribute(b.src)});a.removeAttribute(b.src);c.onerror=function(){b.error&&b.error(a,"invalid");a.className=a.className+" "+b.errorClass};c.onload=function(){"img"===a.nodeName.toLowerCase()?a.src=d:a.style.backgroundImage='url("'+d+'")';a.className=a.className+" "+b.successClass;b.success&&b.success(a)};c.src=d}else b.error&&b.error(a,"missing"),a.className=a.className+" "+b.errorClass}function q(a){e.bottom=(window.innerHeight||document.documentElement.clientHeight)+a;e.right=(window.innerWidth||document.documentElement.clientWidth)+a}function l(a,c,b){a.attachEvent?a.attachEvent&&a.attachEvent("on"+c,b):a.addEventListener(c,b,!1)}function m(a,c,b){a.detachEvent?a.detachEvent&&a.detachEvent("on"+c,b):a.removeEventListener(c,b,!1)}function k(a,c){if(a&&c)for(var b=a.length,d=0;d<b&&!1!==c(a[d],d);d++);}function r(a,c,b){var d=0;return function(){var g=+new Date;g-d<c||(d=g,a.apply(b,arguments))}}var n,e,p;return function(a){if(!document.querySelectorAll){var c=document.createStyleSheet();document.querySelectorAll=function(a,b,d,e,f){f=document.all;b=[];a=a.replace(/\[for\b/gi,"[htmlFor").split(",");for(d=a.length;d--;){c.addRule(a[d],"k:v");for(e=f.length;e--;)f[e].currentStyle.k&&b.push(f[e]);c.removeRule(0)}return b}}var b=this,d=b._util={};d.images=[];d.destroyed=!0;b.options=a||{};b.options.error=a.error||!1;b.options.offset=a.offset||100;b.options.success=a.success||!1;b.options.selector=a.selector||".b-lazy";b.options.separator=a.separator||"|";b.options.container=a.container?document.querySelectorAll(a.container):!1;b.options.errorClass=a.errorClass||"b-error";b.options.breakpoints=a.breakpoints||!1;b.options.successClass=a.successClass||"b-loaded";b.options.src=n=a.src||"data-src";p=1<window.devicePixelRatio;e={};e.top=0-b.options.offset;e.left=0-b.options.offset;b.revalidate=function(){h(this)};b.load=function(a,b){-1===(" "+a.className+" ").indexOf(" "+this.options.successClass+" ")&&t(a,b,this.options)};b.destroy=function(){var a=this._util;this.options.container&&k(this.options.container,function(b){m(b,"scroll",a.validateT)});m(window,"scroll",a.validateT);m(window,"resize",a.validateT);m(window,"resize",a.saveViewportOffsetT);a.count=0;a.images.length=0;a.destroyed=!0};d.validateT=r(function(){f(b)},25,b);d.saveViewportOffsetT=r(function(){q(b.options.offset)},50,b);q(b.options.offset);k(b.options.breakpoints,function(a){if(a.width>=window.screen.width)return n=a.src,!1});h(b)}}
https://requirejs.org/docs/errors.html#mismatch
    at makeError (require-min.js:5)
    at T (require-min.js:5)
    at require-min.js:5
    at script.js:1
makeError @ require-min.js:5
T @ require-min.js:5
(anonymous) @ require-min.js:5
(anonymous) @ script.js:1
setTimeout (async)
Fd.iw.<computed> @ script.js:1
req.nextTick @ require-min.js:5
s @ require-min.js:5
requirejs @ require-min.js:5
(anonymous) @ require-min.js:5
(anonymous) @ require-min.js:5

**Comment by spormeon** [comment]
yes, they all load in order and it works overall now.  I'll look at shim , not holding my breath i'll get it working though

**Comment by ddziaduch** [comment]
Do you load any other JavaScript files? https://requirejs.org/docs/errors.html#mismatch

**Comment by spormeon** [comment]
yes i do,  basically i have a html page, which is a website page and what i'm trying to do is pull in prebid, split out into modules, so i can control , change things easier.  Are you on an IM, as I could show you what i'm trying to do ( its behind logins etc)

**Comment by spormeon** [comment]
one file/ module that wont load is this one,  i'm not sure why,  it loads if i lnk directly to the file, but not if going through require-min.js ?

(slotsonpage.js)
//shows the slots on the page
window.addEventListener("DOMContentLoaded",function(){
document.querySelector("#custom_html-4").insertAdjacentHTML('afterbegin','<div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid8Slot"></div></div>');
document.querySelector("#custom_html-2").insertAdjacentHTML('afterbegin','<div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid11Slot"></div></div><br>');
document.querySelector("#custom_html-3").insertAdjacentHTML('afterbegin','<div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid12Slot"></div></div><br>');
document.querySelector("#anchor_ad").insertAdjacentHTML('afterbegin','<div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid13Slot"></div></div>');
document.querySelector("#custom_html-5").insertAdjacentHTML('afterbegin','<br><div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid4Slot"></div></div><br>');
document.querySelector("#custom_html-7").insertAdjacentHTML('afterbegin','<br><div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid9Slot"></div></div><br>');
document.querySelector("#custom_html-19").insertAdjacentHTML('afterbegin','<br><div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid5Slot"></div></div>');
document.querySelector(".post-the-content p:nth-child(6)").insertAdjacentHTML('afterbegin','<br><br><div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid6Slot"></div></div><br>');
document.querySelector(".post-the-content p:nth-child(10)").insertAdjacentHTML('afterbegin','<br><div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid10Slot"></div></div><br>');
document.querySelector(".comments_headline").insertAdjacentHTML('afterend','<br><br><div class="ad-reporter-ahytrfg35423"><div id="advertisement" style="border: 0pt none; margin: auto; text-align: center; color: #999; text-transform: uppercase; font-family: sans-serif; font-size: 9px; font-weight: 400; letter-spacing: .2em; line-height: 1; margin-top: 0px; position: relative; top: -4px;">Advertisement</div><div id="inreedvid7Slot"></div></div><br>');
});


where as this works and shows the slots on page:
<script rel="preload" type="text/javascript" src="//xxxxxx.com/pub000001/app/slotsonpage.js"></script>
<script rel="preconnect" data-main="pub000001/app" src="https://xxxxxx.com/pub000001/require-min.js"></script>

**Comment by ddziaduch** [comment]
@spormeon maybe it loads but it load **after** the DomContentLoaded so that is why it is not executed. Bear in mind that RequireJS is **async**
